integration-cli/docker_cli_port_test.go

```diff
@@ -221,3 +223,84 @@ func (s *DockerSuite) TestPortExposeHostBinding(c *check.C) {
 		c.Error("Port is still bound after the Container is removed")
 	}
 }
+
+func stopRemoveContainer(id string, c *check.C) {
+	runCmd := exec.Command(dockerBinary, "rm", "-f", id)
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		c.Fatal(out, err)
+	}
+}
+
+func (s *DockerSuite) TestUnpublishedPortsInPsOutput(c *check.C) {
+	// Run busybox with command line expose (equivalent to EXPOSE in image's Dockerfile) for the following ports
+	port1 := 80
+	port2 := 443
+	expose1 := fmt.Sprintf("--expose=%d", port1)
+	expose2 := fmt.Sprintf("--expose=%d", port2)
+	runCmd := exec.Command(dockerBinary, "run", "-d", expose1, expose2, "busybox", "sleep", "5")
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		c.Fatal(out, err)
+	}
+	id := strings.TrimSpace(out)
+
+	// Check docker ps o/p for last created container reports the unpublished ports
+	unpPort1 := fmt.Sprintf("%d/tcp", port1)
+	unpPort2 := fmt.Sprintf("%d/tcp", port2)
+	runCmd = exec.Command(dockerBinary, "ps", "-n=1")
+	out, _, err = runCommandWithOutput(runCmd)
+	if err != nil {
+		c.Fatal(out, err)
+	}
+	if !strings.Contains(out, unpPort1) || !strings.Contains(out, unpPort2) {
+		c.Errorf("Missing unpublished ports(s) (%s, %s) in docker ps output: %s", unpPort1, unpPort2, out)
+	}
+	stopRemoveContainer(id, c)
```

Comment:
yeah it is, and indeed every test clear every container created on test teardown so you don't need to worry about this :) unless of course you really need to delete the container to continue testing